public_app/src/components/Registration/AddMember/FormPersonalDetails.js

```diff
@@ -179,13 +164,49 @@ export const FormPersonalDetails = ({ setValue, formData, navigation, verifyDeta
                 alert("Error while registering, please try again later.\n" + err);
             });
     };
+
+    const updateFormData = (kycData) => {
+        let selectedDistrict = null;
+        stateAndDistricts.states.forEach(state => {
+            if(state.name === kycData.state) {
+                state.districts.forEach(district => {
+                    if(kycData.district !== "" && district.name.indexOf(kycData.district) >= 0) {
+                        selectedDistrict = district.name;
```

Comment:
could we please do a devbox on this? We should be able to do this in a simpler manner

Reply:
it needs two loops to get to correct district
at best what we could do is refactor this to look better